client.go

```diff
@@ -46,8 +54,19 @@ func NewClient(api plugin.API) *Client {
 		Plugin:        PluginService{api: api},
 		Post:          PostService{api: api},
 		Session:       SessionService{api: api},
+		Store:         getStore(api),
 		System:        SystemService{api: api},
 		Team:          TeamService{api: api},
 		User:          UserService{api: api},
 	}
 }
+
+func getStore(api plugin.API) *StoreService {
```

Comment:
The only negative consequence of letting users create multiple clients is it'll reserve connections on the server more than it needs to, and may starve unintentionally, this just prevents that from happening.

Testing is not a huge pain since there is a way to test the store, but you're right, it's not as nice as simple as testing other services.

0/5 either way.

Reply:
Got it -- propose we simplify and document and revisit if this comes up as a concrete concern. All usage to date ends up with a single client per plugin, so I'm hoping it won't be an issue.

Reply:
@lieut-data done. Took out all the singleton code from the client.

Reply:
documented on the client: https://github.com/mattermost/mattermost-plugin-api/pull/46/files#diff-d0b7f86cea13f90d3cc0cc2338da8d5fR33